plugins/hypervisors/kvm/src/main/java/org/apache/cloudstack/kvm/ha/KVMHAProvider.java

```diff
@@ -84,18 +84,24 @@ public boolean recover(Host r) throws HARecoveryException {
 
     @Override
     public boolean fence(Host r) throws HAFenceException {
-
         try {
-            if (outOfBandManagementService.isOutOfBandManagementEnabled(r)){
-                final OutOfBandManagementResponse resp = outOfBandManagementService.executePowerOperation(r, PowerOperation.OFF, null);
-                return resp.getSuccess();
+            // host exists and is managed OOB
+            if (r != null && outOfBandManagementService.isOutOfBandManagementEnabled(r)) {
+                // check host status
+                if (Host.Status.DOWN.equals(r.getStatus())) {
+                    LOG.info("Host " + r.getName() + " is already down. Returning success.");
+                    return true;
+                } else {
+                    final OutOfBandManagementResponse resp = outOfBandManagementService.executePowerOperation(r, PowerOperation.OFF, null);
+                    return resp.getSuccess();
+                }
             } else {
-                logger.warn("OOBM fence operation failed for this host " + r.getName());
+                LOG.warn("OOBM fence operation failed for this host " + r.getName());
```

Comment:
@leo79901 
should it be `logger` instead of `LOG` ?

Reply:
Yes , you are right.